docs/development/API/Dshot.md

```diff
@@ -161,50 +164,56 @@ value = 100000101100
 (&) = 000000000110 # CRC
 ```
 
-So the two bytes transmitted from flight-controller to ESC would be:
+So the two bytes transmitted from the FC to ESC would be:
 
 ```
 1000001011000110
 ```
 
-We would put this signal on the wire to transmit the Dshot frame:
+We would put this signal on the wire to transmit the DShot frame:
 
-![Dshot frame](./dshot_frame.webp)
+![DShot frame](./dshot_frame.webp)
 
-The green part are the throttle bits, blue is the telemetry bit and yellow the CRC checksum.
+The green parts are the throttle bits, blue is the telemetry bit and yellow the CRC checksum.
 
 ### Arming sequence
 
-The arming sequenc might differ from implementation to implementation, but on most firmwares a 0 command is expected for a certain amount of time - 300ms on Bluejay for example - before the ESC goes into a state where it will accept actual throttle values.
+The arming sequence might differ from implementation to implementation, but on most firmwares a 0 command is expected for a certain amount of time - 300ms on Bluejay for example - before the ESC goes into a state where it will accept actual throttle values.
 
 ## Why is frame length important?
 
 The frame length is important because it indicates how fast the ESC can be updated. The shorter the frame length, the more often a frame can be sent per second. In other words the higher the bitrate, the more data we can send per second.
 
 This is then only limited by the loop speed of the flight-controller. Or the other way around, as we will see.
 
-> Let’s have a look at Dshot 300: A frame length of 106.72µs allows us to theoretically send 18768 full frames per second. Which results in a maximum frequency of around 18kHz.
+> Let’s have a look at DShot 300: A frame length of 106.72µs allows us to theoretically send 18768 full frames per second. Which results in a maximum frequency of around 18 kHz.
 
```

Comment:
It would seem that it’s quite ‘tight’ for DShot 300 at 8k, since the DShot frame takes 108us to send, and we only have 125us available. The way the scheduler works, some loops will be shorter than 125us, e.g. when catching up on a previously late loop. This may cause a dropped DShot 300 packet. Hence it may be best to run DShot 600 with 8k PID loops, to ensure that frames are not dropped. 

with a 4k PID loop, we have 250us in which to send the data. Either DShit 300 or DShot 300 may be used. DShot 600 may be less resistant to external noise than DShot 300.

Reply:
Below this quote the text then states:
```
But this is actually not the whole truth, since the flight controller spaces out the frames and locks it to the PID loop frequency. Dshot frame generation thus always runs at PID loop rate - this on the other hand means, that if you are running really high PID loop frequencies, you also need to run a high Dshot version.
```
So dshot300 is not sufficient for 8K due to the timing requirements.

Reply:
Seems to be a typo here. The frame length of 106.72 us is for DShot150. DShot300 has a frame length of 53.28 us, which gives a max update rate of about 18 kHz. 

Note that *bidirectional* DShot300 is not sufficient for 8k, because the rate is halved in bidirectional mode, and there is an extra 30 us break between DShot and telemetry frames.  So the max effective rate for bidirectional DShot300 is about 6 kHz, which is obviously lower than an 8k PID loop rate.

I'll confirm the details with stylesuxx and submit a separate PR to clear this up.

Reply:
Thanks, @ot0tot, for that clarification.  Very important.

We need to have a very clear, coherent basis for our recommendations regarding DShot 300 vs 600 in relation to PID loop rate.